backend/migrator/migrator.go

```diff
@@ -129,10 +134,79 @@ func MigrateSchema(ctx context.Context, connCfg dbdriver.ConnectionConfig, stric
 	return &verAfter, nil
 }
 
-func initializeSchema(ctx context.Context, metadataDB *sql.DB, bytebaseDriver *pg.Driver, cutoffSchemaVersion semver.Version, databaseName, serverVersion string) error {
+func backfillHistory(ctx context.Context, storeInstance *store.Store, bytebasePgDriver *pg.Driver, databaseName string) error {
+	histories, err := storeInstance.ListInstanceChangeHistory(ctx, &store.FindInstanceChangeHistoryMessage{
+		// Metadata database has instanceID nil;
+		InstanceID: nil,
+	})
+	if err != nil {
+		return err
+	}
+	// For new database and backfilled database, there should be histories already.
+	if len(histories) > 0 {
+		return nil
+	}
+
+	log.Info("Backfilling Bytebase metadata migration history")
+	limit := 10
+	offset := 0
+	for {
+		oldHistories, err := bytebasePgDriver.FindMigrationHistoryList(ctx, &dbdriver.MigrationHistoryFind{
+			Database: &databaseName,
+			Limit:    &limit,
+			Offset:   &offset,
+		})
+		if err != nil {
+			return err
+		}
+		if len(oldHistories) == 0 {
+			break
+		}
+		offset += limit
+
+		var creates []*store.InstanceChangeHistoryMessage
+		for _, h := range oldHistories {
+			storedVersion, err := util.ToStoredVersion(h.UseSemanticVersion, h.Version, h.SemanticVersionSuffix)
+			if err != nil {
+				return err
+			}
+			changeHistory := store.InstanceChangeHistoryMessage{
+				CreatorID:           api.SystemBotID,
+				CreatedTs:           h.CreatedTs,
+				UpdaterID:           api.SystemBotID,
+				UpdatedTs:           h.UpdatedTs,
+				InstanceID:          nil,
+				DatabaseID:          nil,
+				IssueID:             nil,
+				ReleaseVersion:      h.ReleaseVersion,
+				Sequence:            int64(h.Sequence),
+				Source:              h.Source,
+				Type:                h.Type,
+				Status:              h.Status,
+				Version:             storedVersion,
+				Description:         h.Description,
+				Statement:           h.Statement,
+				Schema:              h.Schema,
+				SchemaPrev:          h.SchemaPrev,
+				ExecutionDurationNs: h.ExecutionDurationNs,
+				Payload:             h.Payload,
+			}
+
+			creates = append(creates, &changeHistory)
+		}
+
+		if _, err := storeInstance.CreateInstanceChangeHistory(ctx, creates...); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func initializeSchema(ctx context.Context, storeInstance *store.Store, metadataDriver dbdriver.Driver, cutoffSchemaVersion semver.Version, serverVersion string) error {
 	// We use instance_change_history table to determine whether we've initialized the schema.
 	var exists bool
-	if err := metadataDB.QueryRowContext(ctx,
+	if err := metadataDriver.GetDB().QueryRowContext(ctx,
 		`SELECT EXISTS (SELECT 1 FROM information_schema.tables WHERE table_name = 'instance_change_history')`,
```

Comment:
What will happen if I upgrade an older version of bytebase which has its schema initialized but doesn't have the instance_change_history table.

Reply:
Good finding: https://github.com/bytebase/bytebase/pull/5449